pyi.py

```diff
@@ -159,7 +159,41 @@ class TypeVarInfo(NamedTuple):
 )
 
 
+class PyflakesPreProcessor(ast.NodeTransformer):
+    """Transform AST prior to passing it to pyflakes.
+
+    This reduces false positives on recursive class definitions.
+    """
+
+    def visit_ClassDef(self, node: ast.ClassDef) -> ast.ClassDef:
+        self.generic_visit(node)
+        node.bases = [
+            # Remove the subscript to prevent F821 errors from being raised
+            # for (valid) recursive definitions.
```

Comment:
I spent a while staring at this, so maybe a comment is helpful:

```suggestion
            # Remove the subscript to prevent F821 errors from being raised
            # for (valid) recursive definitions: Foo[Bar] --> Foo
```